src/include/btmem.h

```diff
@@ -856,6 +857,24 @@ struct __wt_ref {
 #define	WT_REF_SPLIT	 7		/* Parent page split (WT_REF dead) */
 	volatile uint32_t state;	/* Page state */
 
+	struct {
+		WT_SESSION_IMPL *session;
+		const char *name;
+		const char *file;
+		int line;
+		uint32_t state;
+	} hist[3];
+	int histoff;
```

Comment:
My only minor comment about the `WT_REF_SETSTATE` macro is trying to find a way to enforce its usage. How difficult would it be to look for `ref->state = ` in one of the dist scripts to flag it perhaps? That should not hold up the merge, just an idea.

Reply:
Yes, we have checks like this in `s_style`: adding something like this (and the `WT_PUBLISH` equivalent) makes sense to me.

Reply:
@sueloverso, @michaelcahill 

I think that's going to be difficult to do, it's easy to split the `=` operator from the `WT_REF_XXX` name, and we have places we set `WT_REF.state` in interesting ways. For example the `new_state` and `final_statep` uses:

```
% egrep ' = WT_REF_' ../*/*.[chyis]
../btree/bt_handle.c:	root_ref->state = WT_REF_MEM;
../btree/bt_read.c:		*final_statep = WT_REF_LIMBO;
../btree/bt_read.c:		new_state = WT_REF_READING;
../btree/bt_read.c:		new_state = WT_REF_LOCKED;
../btree/bt_read.c:	final_state = WT_REF_MEM;
../btree/bt_rebalance.c:	copy->state = WT_REF_DISK;
../btree/bt_split.c:	child->state = WT_REF_MEM;
../btree/bt_split.c:	child->state = WT_REF_MEM;
../evict/evict_lru.c:	*previous_statep = WT_REF_MEM;
../evict/evict_page.c:			child->state = WT_REF_DELETED;
```

An alternative would be to obscure the field name, and add `WT_REF_STATE_READ` and `WT_REF_STATE_WRITE` macros.